core/src/main/java/nu/yona/server/messaging/entities/Message.java

```diff
@@ -20,7 +21,11 @@
 @Table(name = "MESSAGES")
 public abstract class Message extends EntityWithID
 {
+	@Column(nullable = false)
+	private UUID destinationID;
```

Comment:
If you make the relationship between ``Message`` and ``MessageDestination`` two way navigable, you can still easily find messages for a destination, without having to abandon the relationship in the domain model.